components/Header.jsx

```diff
@@ -0,0 +1,303 @@
+          e.preventDefault()
+          setSecondaryMenuItemIndex(0)
+        }
+      },
+      onClick: () => {
+        if (isActive) {
+          setSecondaryMenuItemIndex(isDropdownClosed ? 0 : null)
+        } else {
+          setPrimaryMenuItemIndex(itemIndex)
+          setSecondaryMenuItemIndex(0)
+        }
+      },
+    }
+  }
+  const bindSecondaryMenu = () => ({ role: "menu" })
+  const bindSecondaryMenuItem = (parentIndex, itemIndex) => {
+    const isSelectable =
+      parentIndex === primaryMenuItemIndex &&
+      itemIndex === secondaryMenuItemIndex
+    return {
+      tabIndex: isSelectable ? 0 : -1,
+      onKeyDown: (e) => {
+        if (e.key === "Escape") {
+          setSecondaryMenuItemIndex(null)
+        }
+      },
+      role: "menuitem",
+    }
+  }
+
+  return {
+    mobileMenuOpen,
+    menuBarHasFocus,
+    primaryMenuItemIndex,
+    secondaryMenuItemIndex,
+    bindToggle,
+    bindPrimaryMenu,
+    bindPrimaryMenuItem,
+    bindSecondaryMenu,
+    bindSecondaryMenuItem,
+  }
+}
+
+/**
+ * @see https://www.w3.org/WAI/ARIA/apg/example-index/disclosure/disclosure-navigation-hybrid.html
+ */
+const Header = () => {
+  const [pageScrolled, setPageScrolled] = useState(false)
+
+  const navigationItems = [
+    {
+      value: "/apps",
+      label: <FormattedMessage id="nav.apps" defaultMessage="Apps" />,
+    },
+    {
+      value: "/sponsors",
+      label: <FormattedMessage id="nav.sponsors" defaultMessage="Sponsors" />,
+    },
+    {
+      value: "https://github.com/mastodon/mastodon",
+      label: <FormattedMessage id="nav.code" defaultMessage="Code" />,
+    },
+    {
+      key: "resources",
+      label: <FormattedMessage id="nav.resources" defaultMessage="Resources" />,
+      childItems: [
+        {
+          value: "https://blog.joinmastodon.org/",
+          label: <FormattedMessage id="nav.blog" defaultMessage="Blog" />,
+        },
+        {
+          value: "https://github.com/mastodon/mastodon/discussions",
+          label: <FormattedMessage id="nav.support" defaultMessage="Support" />,
+        },
+        {
+          value: "https://docs.joinmastodon.org",
+          label: (
+            <FormattedMessage id="nav.docs" defaultMessage="Documentation" />
+          ),
+        },
+      ],
+    },
+    {
+      key: "locale",
+      label: <>文A</>,
+      childItems: locales.map((locale) => ({
+        key: locale.code,
+        locale: locale.code,
+        value: "", // current page
+        label: locale.language,
+      })),
+    },
+  ]
+
+  const {
+    mobileMenuOpen,
+    menuBarHasFocus,
+    primaryMenuItemIndex,
+    secondaryMenuItemIndex,
+    bindToggle,
+    bindPrimaryMenu,
+    bindPrimaryMenuItem,
+    bindSecondaryMenu,
+    bindSecondaryMenuItem,
+  } = useMenu({ navigationItems })
+
+  const checkPageScroll = (e) => {
+    if (window.scrollY > 0) {
+      setPageScrolled(true)
+    } else {
+      setPageScrolled(false)
+    }
+  }
```

Comment:
can this just be:
```
  const checkPageScroll = () => {
      setPageScrolled(window.scrollY > 0)
  }
```

since `window.scrollY > 0` will be true or false anyway?